qiskit/dagcircuit/dagdependency.py

```diff
@@ -0,0 +1,502 @@
+    """
+
+    def __init__(self):
+        """
+        Create an empty DAGDependency.
+        """
+        # Circuit name
+        self.name = None
+
+        # Directed multigraph whose nodes are operations(gates) and edges
+        # represent non-commutativity between two gates.
+        self._multi_graph = rx.PyDAG()
+
+        # Map of qreg name to QuantumRegister object
+        self.qregs = OrderedDict()
+
+        # Map of creg name to ClassicalRegister object
+        self.cregs = OrderedDict()
+
+    def to_networkx(self):
+        """Returns a copy of the DAGDependency in networkx format."""
+        # For backwards compatibility, return networkx structure from terra 0.12
+        # where DAGNodes instances are used as indexes on the networkx graph.
+
+        dag_networkx = nx.MultiDiGraph()
+
+        for node in self.get_nodes():
+            dag_networkx.add_node(node['operation']._node_id, operation=node['operation'],
+                                  successors=node['successors'], predecessors=node['predecessors'])
+        for node in self.topological_nodes():
+            for source_id, dest_id, edge in \
+                    self.get_in_edges(node['operation']._node_id):
+                dag_networkx.add_edge(source_id, dest_id, **edge)
+        return dag_networkx
+
+    def to_retworkx(self):
+        """ Returns the DAGDependency in retworkx format."""
+        return self._multi_graph
+
+    def qubits(self):
+        """Return a list of qubits (as a list of Qubit instances)."""
+        return [qubit for qreg in self.qregs.values() for qubit in qreg]
+
+    def clbits(self):
+        """Return a list of classical bits (as a list of Clbit instances)."""
+        return [clbit for creg in self.cregs.values() for clbit in creg]
+
+    def size(self):
+        """ Returns the number of gates in the circuit"""
+        return len(self._multi_graph)
+
+    def depth(self):
+        """Return the circuit depth.
+        Returns:
+            int: the circuit depth
+        """
+        depth = rx.dag_longest_path_length(self._multi_graph)
+        return depth if depth >= 0 else 0
+
+    def add_qreg(self, qreg):
+        """Add qubits in a quantum register."""
+        if not isinstance(qreg, QuantumRegister):
+            raise DAGDependencyError("not a QuantumRegister instance.")
+        if qreg.name in self.qregs:
+            raise DAGDependencyError("duplicate register %s" % qreg.name)
+        self.qregs[qreg.name] = qreg
+
+    def add_creg(self, creg):
+        """Add all wires in a classical register."""
+        if not isinstance(creg, ClassicalRegister):
+            raise DAGDependencyError("not a ClassicalRegister instance.")
+        if creg.name in self.cregs:
+            raise DAGDependencyError("duplicate register %s" % creg.name)
+        self.cregs[creg.name] = creg
+
+    def _add_multi_graph_node(self, node):
+        """
+        Args:
+            node (DAGNode): considered node.
```

Comment:
This is not the right argument type, as we are not adding a `DAGNode` here, but a dictionary of 
```
{'operation': <DAGNode>, 'successors': [], 'predecessors': []}
```
maybe we should think about a separate class for the nodes of `DAGDependency` which are distinct from the usual `DAGNode` used for the `DAGCircuit`?

The current DAGNode is class consisting of the following attributes:
```
type, op, name, qargs, cargs, condition
```

The new node could have the following attributes:
```
op, name, qargs, cargs, condition, successors, predecessors
```
(no need for `type` as we always have 'op' types)

What do you think?